lib/rules/array-bracket-spacing.js

```diff
@@ -179,8 +179,10 @@ module.exports = {
 
             const first = sourceCode.getFirstToken(node),
                 second = sourceCode.getFirstToken(node, 1),
-                penultimate = sourceCode.getLastToken(node, 1),
-                last = sourceCode.getLastToken(node),
+                last = node.typeAnnotation
+                    ? sourceCode.getTokenBefore(node.typeAnnotation)
```

Comment:
I wanted to do that – the problem is that `node.elements` is an empty array if there are no elements, so there may not be a last element. It turns out that

```js
([]: any) => {}
```

is actually legal. Deeply silly, yes, but actually a thing.